src/domain/service/note.ts

```diff
@@ -441,4 +449,40 @@ export default class NoteService {
 
     return noteHistoryPublic;
   }
+
+  /**
+   * Get note parent structure recursively by note id and user id
+   * and check if user has access to the parent note.
+   * @param noteId - id of the note to get parent structure
+   * @param userId - id of the user that is requesting the parent structure
+   * @returns - array of notes that are parent structure of the note
+   */
+  public async getNoteParentStructure(noteId: NoteInternalId, userId: number | null): Promise<Array<{ noteId: Note['publicId']; content: Note['content'] }>> {
+    const noteUserAccess = await this.teamRepository.getTeamMemberByNoteAndUserId(userId as NoteInternalId, noteId);
+
+    if (userId === null || noteUserAccess === null) {
+      const nextNoteId = await this.getParentNoteIdByNoteId(noteId);
+
```

Comment:
if userId is null, then no need to check for parent note teams 
(any team can not have user with id null in it)